### ISSUE TITLE
Why CSS templates are not applied in Angular applications?

### ISSUE THREAD

**Issue by izaera** [liferay/liferay-npm-build-tools#158]
Does somebody know where to put (s)css files within a npm-angular-portlet? I tried to put them into `...js/styles` and reference them like template `/o/my-portlet/js/styles/my.css` but this doesn't work.
While i can access the file the styles are not applied, the classes are not visible.

**Comment by izaera** [comment]
The problem is that Angular has a bug that hasn't been fixed for long and prevents developers from using absolute URLs for CSS templates. More info in: 

- https://aigeec.com/angularjs-2-style-urls/
- https://github.com/angular/angular/issues/4974

Sadly, there's no possible bypass, because using relative URLs is not an option in a portal environment because you don't know the URL in advance and, in fact, it can change depending on where you put your portlet so, a relative URL that works for one page may not work for another 😞.

**What can be done then?**

Well, our recomendation is to move all CSSs to a theme/themelet and deploy it. This is a better approach than structuring the CSSs by component because with themes the styles can be changed, overriden, are better maintained,... 

In general, using themes fits portal's philosophy better.

**Comment by MarcoPortillo** [comment]
@izaera what if the CSS file that is not working comes from a node_modules package? My own css files are working if I just import them in the com.liferay.portlet.header-portlet-css (/css/styles.css) but one that is being use from a npm package will always have problems, what to do there? I already change to different libraries and all have the same problem, all of them use css files

**Comment by izaera** [comment]
@MarcoPortillo we added webpack-like "loaders" in #303 for that.

There used to be an example of use in [this repo](https://github.com/izaera/liferay-js-toolkit-showcase/commits/loaders), but I broke it some days ago :-(. I have to fix it but I need some spare time.

However, adding a `rule` like [this](https://github.com/izaera/liferay-js-toolkit-showcase/commit/e3011cb022c9110c909eb129361e9ada4c3ea986#diff-42d79c46a700780eb0a5e16b605423b8R30-R34) should make all third party CSS files available as "magic" JS modules that load the CSS.

Have  a look at [the bundler's documentation](https://github.com/liferay/liferay-js-toolkit/wiki/How-to-use-liferay-npm-bundler#the-two-modes-of-operation) too to understand how everything works.
